frontend/views/containers/proposals/ProposalsWidget.vue

```diff
@@ -134,14 +135,33 @@ export default ({
     }
   },
   methods: {
-    onProposalArchived (hashWithProp) {
-      this.ephemeral.archivedProposals.unshift(hashWithProp)
+    onProposalArchived (groupId, hash, proposal) {
+      if (groupId === this.currentGroupId) {
+        this.ephemeral.archivedProposals.unshift([hash, proposal])
+        this.ephemeral.timeouts.push(setTimeout(() => {
+          this.ephemeral.archivedProposals = this.ephemeral.archivedProposals.filter(x => x[0] !== hash)
+        }, new Date(proposal.dateClosed).getTime() - Date.now() + DAYS_MILLIS))
+      }
+    },
+    async refreshArchivedProposals () {
+      const key = `proposals/${this.ourUsername}/${this.currentGroupId}`
+      const archivedProposals = await sbp('gi.db/archive/load', key) || []
+      // proposals which are archived in the last 24 hours
+      this.ephemeral.archivedProposals = archivedProposals
+        .filter(([hash, prop]) => Date.now() - new Date(prop.dateClosed).getTime() < DAYS_MILLIS)
+        .sort(([hash1, prop1], [hash2, prop2]) => new Date(prop2.dateClosed).getTime() - new Date(prop1.dateClosed).getTime())
       // after a day, remove it from the list
-      this.ephemeral.timeouts.push(setTimeout(() => {
-        this.ephemeral.archivedProposals = this.ephemeral.archivedProposals.filter(x => {
-          return x[0] !== hashWithProp[0]
-        })
-      }, 1000 * 60 * 60 * 24)) // 1 day
+      this.clearTimeouts()
+      for (const [hash, proposal] of this.ephemeral.archivedProposals) {
+        this.ephemeral.timeouts.push(setTimeout(() => {
+          this.ephemeral.archivedProposals = this.ephemeral.archivedProposals.filter(x => x[0] !== hash)
+        }, new Date(proposal.dateClosed).getTime() - Date.now() + DAYS_MILLIS))
+      }
```

Comment:
I do not understand what's going on here.

- Why are archived proposals being loaded?
- Why is there a DRY violation on lines 156 - 158?

This widget needs to just show the active open proposals. There is no need for it to touch archived proposals by calling `'gi.db/archive/load'` as far as I can tell.

The only reason `this.ephemeral.archivedProposals` exists is so that we can remember to remove recently archived proposals from the list.

If we load up a group for the first time, we do not need to display recently archived proposals at all. Only if they're *already displayed* do we manually hide them after a day.

Reply:
The Figma design describes [here](https://www.figma.com/file/mxGadAHfkWH6qApebQvcdN/Group-Income-2.0?type=design&node-id=22920-209940&mode=design&t=nu63dKARoxrMur2H-4) that the closed proposals should be displayed in the dashboard widget for 24 hours.
@taoeffect, can you please check this Figma again, and confirm that the above workflow is correct?

Reply:
Sorry, you're right. This code looks OK.